drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,207 @@
+
+Version 3 transaction format will be introduced for Overwinter.
+
+The version field must have its most significant bit set.
+
+There will be new fields for:
+
+* branch id
+* expiry height
+
+======== =============== =========================== =======
+Version  Field           Description                 Type
+======== =============== =========================== =======
+>= 3     version         must have MSB set           int32
+>= 3     branch_id       branch/fork identifier      uint32
+>= 3     expiry_height   block height                uint32
+>= 1     in_count        varint                      1-9 bytes
+>= 1     tx_inputs       list of inputs              vector
+>= 1     out_count       varint                      1-9 bytes
+>= 1     tx_outputs      list of outputs             vector
+>= 1     lock_time       block height or timestamp   uint32
+>= 2     nJoinSplit      varint                      1-9 bytes
+>= 2     vJoinSplit      list of joinsplits          vector
+>= 2     joinSplitPubKey joinSplitSig public key     32 bytes
+>= 2     joinSplitSig    signature                   64 bytes
+======== =============== =========================== =======
+
+
+Transaction Version
+-------------------
+
+The version field is always serialized in little-endian format.
+
+Version 1 transaction 5c6ba844e1ca1c8083cd53e29971bd82f1f9eea1f86c1763a22dd4ca183ae061
+
+* begins with 0x01000000
+* 32-bit signed integer 00 00 00 01 == 1
+
+Version 2 transaction 4435bf8064e74f01262cb1725fd9b53e600fa285950163fd961bed3a64260d8b
+
+* begins with 0x02000000
+* 32-bit signed integer 00 00 00 02 == 2
+
+Legacy parsers require the transaction version number to be positive.
+
+Overwinter parsers also require the transaction version number to be positive, however the serialized version field must have the most significant bit set.
+
+By setting the most significant bit of the version field, we ensure there is replay protection between Legacy and Overwinter compatible software.  With two's complement integers, the most significant bit indicates whether an integer is positive or negative, therefore Overwinter version fields will be negative.
+
+Consider the following example:
+
+When serialized, a Version 3 Overwinter transaction will take the form:
+
+* Little endian format: 0x03000080
+* Representing a 32-bit signed integer: 0x80000003
+* Which has a decimal value of: -2147483645
+
+Legacy parsers will expect the version to be a positive value, such as 1 or 2, and will thus reject the Overwinter transaction as invalid.
+
+Overwinter parsers will accept the transaction as valid as the most significant bit of the 32-bit signed integer has been set.  By masking off (unsetting) the most significant bit, the parser can retrieve the actual transaction version number::
+
+    0x80000003 & 0x7FFFFFFFF = 0x00000003 = 3
+
+Existing code can continue to check the transaction version using greater than comparison operators::
+
+    if (tx.nVersion >= 3) {
```

Comment:
That's "greater than or equal to", not "greater than". Maybe just say "comparison operators".